cpp/src/arrow/python/arrow_to_pandas.cc

```diff
@@ -1424,7 +1479,11 @@ class DataFrameBlockCreator {
     for (int i = 0; i < table_->num_columns(); ++i) {
       std::shared_ptr<ChunkedArray> col = table_->column(i);
       PandasBlock::type output_type = PandasBlock::OBJECT;
-      RETURN_NOT_OK(GetPandasBlockType(*col, options_, &output_type));
+      if (extension_columns_.count(table_->field(i)->name())) {
```

Comment:
Hmm... I don't understand why we're using an explicit `extension_columns`. Shouldn't we simply detect an arrow ExtensionType?

Reply:
I was also confused by this. I looked at the unit test below and there are a couple of different things going on:

* Creating pandas ExtensionArray values from built-in Arrow types
* Converting Arrow ExtensionType data 

This seems to do the former but not the latter. What is the use case for the former, mainly getting IntegerArray out?